data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/S3ScanPartitionCreationSupplier.java

```diff
@@ -174,14 +176,22 @@ private Instant getMostRecentLastModifiedTimestamp(final ListObjectsV2Response l
     }
 
     private boolean shouldScanBeSkipped(final Map<String, Object> globalStateMap) {
+
         if (Objects.isNull(schedulingOptions) && hasAlreadyBeenScanned(globalStateMap)) {
-            LOG.info("Skipping scan because the buckets have already been scanned once");
+
+            if (!(Boolean) globalStateMap.get(SINGLE_SCAN_COMPLETE)) {
+                LOG.info("Single S3 scan has already been completed");
```

Comment:
Why log only during the seconds attempt at scanning? Why not log once the first scan is complete?

Reply:
We could log after the first scan, but with source coordination this will only get run again after all the scanned objects have been processed, so as far as timing of the log it makes more since when trying to scan again